vj4/handler/record.py

```diff
@@ -69,13 +74,13 @@ class RecordMainConnection(base.Connection):
     rdoc['udoc'], rdoc['pdoc'] = await asyncio.gather(
       user.get_by_uid(rdoc['uid']), problem.get(rdoc['domain_id'], rdoc['pid']))
     # TODO(iceboy): check permission for visibility. (e.g. test).
-    self.send(html=self.render_html('record_tr.html', rdoc=rdoc))
+    self.send(html=self.render_html('record_tr.html', rdoc=rdoc), rdoc=rdoc)
 
   async def on_close(self):
     bus.unsubscribe(self.on_record_change)
 
 
-@app.route('/records/{rid}', 'record_detail')
+@app.route(r'/records/{rid}', 'record_detail')
```

Comment:
shouldn't all routes like `'.../{...}/...'` be prefixed by `r`? If not, I have mis-understood the usage of this flag.